src/main/java/com/example/SurveyController.java

```diff
@@ -92,6 +92,18 @@ public String viewSurveys(Model model) {
         return "viewsurveys";
     }
 
+    @PostMapping("/survey/{id}/activate")
+    public String activateSurvey(@PathVariable("id") Long id) {
+        Optional<Survey> optionalSurvey = surveyRepository.findById(id);
+        if (optionalSurvey.isPresent()) {
+            Survey survey = optionalSurvey.get();
+            survey.setActive(true);
+            surveyRepository.save(survey);
+        }
+        return "redirect:/survey?selectedSurvey=" + id;
```

Comment:
just checking, after the activation/deactivation, we want to show the page with the survey information instead of going to the home page?

Reply:
It'll stay on the same page but refresh it to update the Status and the button